planning/costmap_generator/config/costmap_generator.param.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 UCI SORA Lab
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+/**:
+  ros__parameters:
+    costmap_frame: "map"
+    vehicle_frame: "base_link"
+    map_frame: "map"
+    update_rate: 10.0
+    activate_by_scenario: true
```

Comment:
According to the description ("if true, activate by scenario = parking. Otherwise, activate if vehicle is inside parking lot."), I think the default value should be set by the current scenario the vehicle is in.

I set it to `True` according to the default values of the [previous .cpp file](https://github.com/autowarefoundation/autoware.universe/commit/9952f94e6c37c18c7e78c0c1178de171dbd01b8d), which was already deleted.

Reply:
Ah, I see. Makes sense.

What about updating autoware.universe/launch/tier4_planning_launch/launch/scenario_planning/parking.launch.py?

Reply:
Yes, we need to update the `autoware.universe/launch/tier4_planning_launch/launch/scenario_planning/parking.launch.py` as well.

I've modified that `.py` file. Shall we still keep the `activate_by_scenario = true` in the `launch.xml/schema.json/param.yaml` of the `planning/costmap_generator` directory?

Reply:
The default in the schema is an annotation. It will be used eventually to render the documentation from it, but it doesn't enforce anything in software. So it doesn't matter so much, `true` is fine.

Actually, for the modification in `parking.launch.py` I had in mind of replacing the list of parameters with a reference to the config file of this package instead, in the same way you do for this package's launch file. Not just replacing the value of `activate_by_scenario`.

Reply:
On second thought, I don't know if the tier4 package will want to use the defaults from another package that it doesn't control. Not sure what the best solution is.

@satoshi-ota I think it relates to the point you raise at https://github.com/autowarefoundation/autoware.universe/pull/4699#issuecomment-1700415069. Except here it doesn't break the launch of anything if we leave it as it currently is.

Reply:
I think we'd better keep only one configuration file that controls the node. Having the same set of parameters in two different places would cause some biases or inconsistencies.

Reply:
currently the default value of `activate_by_scenario` was removed and set only in  laucher side https://github.com/autowarefoundation/autoware_launch/pull/679.

And we need to set it `False` for pull over of lane driving scenario.

https://github.com/autowarefoundation/autoware.universe/pull/2879

so could you please set `False`?